plugins/core/jetbrains-community/src/software/aws/toolkits/jetbrains/services/amazonq/FeatureDevSessionContext.kt

```diff
@@ -92,16 +94,24 @@ class FeatureDevSessionContext(val project: Project) {
 
     suspend fun zipFiles(projectRoot: VirtualFile): File = withContext(getCoroutineBgContext()) {
         val files = mutableListOf<VirtualFile>()
+        var totalSize: Long = 0
+
         VfsUtil.visitChildrenRecursively(
             projectRoot,
             object : VirtualFileVisitor<Unit>() {
                 override fun visitFile(file: VirtualFile): Boolean {
-                    if (file.isFile) {
-                        files.add(file)
-                        return true
-                    }
                     return runBlocking {
-                        !ignoreFile(file, this)
+                        if (file.isFile && !ignoreFile(file, this)) {
+                            totalSize += file.length
+                            files.add(file)
+
+                            if (maxProjectSizeBytes != null && totalSize > maxProjectSizeBytes) {
+                                throw RepoSizeLimitError(message("amazonqFeatureDev.content_length.error_text"))
+                            }
+                            return@runBlocking true
+                        } else {
+                            return@runBlocking !ignoreFile(file, this)
```

Comment:
With this implementation, we'll run `ignoreFile(file, this)` function twice for the files that should be ignored. Can we - either store result of `ignoreFile()` and use it here
- or add a check to this else statement that checks if it is not file, then call this function?

Reply:
I can put it in a variable and reuse it

Reply:
Asking because, the ignoreFile function is doing pattern matching based on the regexp and this implementation can increase the duration needed to check files.

Reply:
> Asking because, the ignoreFile function is doing pattern matching based on the regexp and this implementation can increase the duration needed to check files.

Yes will make it slower, by how much not sure. But adding another variable isn't a big deal so i can do it anyway